lib/workers/global/config/parse/file.ts

```diff
@@ -1,8 +1,28 @@
+import { readFile } from 'fs-extra';
+import { load } from 'js-yaml';
+import JSON5 from 'json5';
 import upath from 'upath';
 import { migrateConfig } from '../../../../config/migration';
-import type { AllConfig } from '../../../../config/types';
+import type { AllConfig, RenovateConfig } from '../../../../config/types';
 import { logger } from '../../../../logger';
 
+export async function getParsedContent(file: string): Promise<RenovateConfig> {
+  switch (upath.extname(file)) {
+    case '.yaml':
+    case '.yml':
+      return load(await readFile(file, 'utf8'), {
+        json: true,
+      }) as RenovateConfig;
+    case '.json5':
+      return JSON5.parse(await readFile(file, 'utf8'));
+    default: {
+      // .json and .js
+      const tmpConfig = await import(file);
+      return tmpConfig.default ? tmpConfig.default : tmpConfig;
+    }
+  }
```

Comment:
@viceice initially I thought this switch would be cleaner than if/else clauses but I now do duplicate readfiles to avoid reading it first even on default. Do you think this is ok or is there a cleaner way here?

Also unsure about trying to catch errors here as the calling functions wrap it already differently, but I'll wait for comments :) Thanks for your patience 🐱

Reply:
no error handling here beside my comment above.